synapse/handlers/auth.py

```diff
@@ -1214,33 +1198,43 @@ async def _validate_userid_login(
 
         known_login_type = False
 
-        for provider in self.password_providers:
-            supported_login_types = provider.get_supported_login_types()
-            if login_type not in supported_login_types:
-                # this password provider doesn't understand this login type
-                continue
-
+        # get all of the login_types registered by modules
+        supported_login_types = self.password_auth_provider.get_supported_login_types()
+        # check if the login type being used is supported by a module
+        # note that if LoginType.PASSWORD has been disabled then login_type has already been
+        # checked for that by this point
```

Comment:
So there's a config option that disables password authentication. I mean that we don't need to remove LoginType.PASSWORD from the list types that modules support because by this point in the code, we know that the variable `login_type` is not LoginType.PASSWORD

What would be a better way of phrasing this?